cmake/atomic_configure_file.cmake

```diff
@@ -1,9 +1,9 @@
 function(atomic_configure_file input output)
-  set(atomic_file "${CMAKE_BINARY_DIR}/atomic_configure_file")
+  get_filename_component(atomic_file ${output} NAME)
+  set(atomic_file "${CMAKE_BINARY_DIR}/${atomic_file}")
   configure_file("${input}" "${atomic_file}" ${ARGN})
   get_filename_component(output_path ${output} PATH)
-  if(NOT EXISTS ${output_path})
-    file(MAKE_DIRECTORY ${output_path})
-  endif()
-  file(RENAME "${atomic_file}" "${output}")
+  # sync multiple catkin cmake processes writing to that location
+  file(LOCK "${output_path}" DIRECTORY GUARD FUNCTION)
```

Comment:
This signature is only available as of CMake 3.2. Please use it only conditionally.

While this ensures mutually exclusive access in CMake are any of the files generated by this function accessed in parallel from non-CMake processes?